test-data/unit/check-optional.test

```diff
@@ -162,6 +163,15 @@ x = None
 x = 1
 reveal_type(x)  # E: Revealed type is 'builtins.int'
 
+[case testInferOptionalAnyType]
+from typing import Any
+x = None
+a = None  # type: Any
```

Comment:
Added test cases for non-optional union type and a union type without `Any` items